crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -1004,14 +1013,37 @@ unsafe impl SystemParam for &'_ World {
     }
 }
 
-/// SAFETY: `DeferredWorld` can read all components and resources but cannot be used to gain any other mutable references.
+const MUT_DEFERRED_WORLD_ERROR: &str = "DeferredWorld requires exclusive access
+    to the entire world, but a previous system parameter already registered \
+    access to a part of it. Allowing this would break Rust's mutability rules";
+
+// SAFETY: `DeferredWorld` registers exclusive access to the entire world.
 unsafe impl<'w> SystemParam for DeferredWorld<'w> {
     type State = ();
     type Item<'world, 'state> = DeferredWorld<'world>;
 
     fn init_state(_world: &mut World, system_meta: &mut SystemMeta) -> Self::State {
-        system_meta.component_access_set.read_all();
-        system_meta.component_access_set.write_all();
+        let mut access = Access::default();
+        access.read_all();
+        if !system_meta
```

Comment:
Does the existing code for DeferredWorld not check for conflicts? This change should probably pulled out if so as that's a safety concern.